mono/metadata/w32process.c

```diff
@@ -561,13 +595,19 @@ ves_icall_System_Diagnostics_Process_GetModules_internal (MonoObjectHandle this_
 	return_val_if_nok (error, NULL_HANDLE_ARRAY);
 
 	for (i = 0; i < module_count; i++) {
-		if (mono_w32process_module_get_name (process, mods [i], modname, MAX_PATH)
-			 && mono_w32process_module_get_filename (process, mods [i], filename, MAX_PATH))
-		{
+		modname = mono_w32process_module_get_name (process, mods [i], &modname_len);
+		filename = mono_w32process_module_get_filename (process, mods [i], &filename_len);
+		if (modname && filename) {
 			process_add_module (module, filever, str, process, mods [i], filename, modname, get_process_module_class (), error);
-			return_val_if_nok (error, NULL_HANDLE_ARRAY);
+			if (!is_ok (error)) {
+				g_free (modname);
+				g_free (filename);
+				return NULL_HANDLE_ARRAY;
+			}
 			MONO_HANDLE_ARRAY_SETREF (temp_arr, num_added++, module);
 		}
+		g_free (modname);
+		g_free (filename);
```

Comment:
You don't need to free twice.
Instead:
```
  if (!is_ok (error))
    break;
  array setref
 }
g_free twice
if (!is_ok())
  return
```

Reply:
I don't think the code you suggested will work since they have to be freed each iteration of the loop, but you're right that this can be cleaned up a bit to only free once.

Reply:
The old code stopped at the first error and returned.
So either undo this part, sorry, or try for single point of exit.

Reply:
Check my latest commit and see if you think that's fine, I tried to clean it up.

Reply:
I don’t think so but hard to see in github on phone. The old code stopped the loop upon error. This does not seem to.

Reply:
Oh oops probably. Just hard to see indent on phone.